openxr/src/session.rs

```diff
@@ -308,6 +310,102 @@ impl<G> Session<G> {
         }
     }
 
+    /// Enumerates valid render model paths supported by the runtime that can be used in `Session::get_render_model_properties_fb`.
+    ///
+    /// Requires FB_render_model
+    pub fn enumerate_render_model_paths_fb(&self) -> Result<Vec<Path>> {
+        let raw = get_arr(|cap, count, buf| unsafe {
+            (self
+                .instance()
+                .render_model_fb()
+                .enumerate_render_model_paths)(self.as_raw(), cap, count, buf)
+        })?;
+        Ok(raw.into_iter().map(|path_info| path_info.path).collect())
+    }
+
+    #[inline]
+    /// Gets information for a render model using a path retrived from `Session::enumerate_render_model_paths_fb`
+    ///
+    /// Requires FB_render_model
+    pub fn get_render_model_properties_fb(
+        &self,
+        path: Path,
+        flags: RenderModelFlagsFB,
+    ) -> Result<RenderModelPropertiesFB> {
+        let out = unsafe {
+            let mut capabilities = sys::RenderModelCapabilitiesRequestFB {
+                ty: sys::RenderModelCapabilitiesRequestFB::TYPE,
+                next: ptr::null_mut(),
+                flags,
+            };
+            let mut out = sys::RenderModelPropertiesFB::out(&mut capabilities as *mut _ as _);
+            cvt((self
+                .instance()
+                .render_model_fb()
+                .get_render_model_properties)(
+                self.as_raw(),
+                path,
+                out.as_mut_ptr(),
+            ))?;
+            out.assume_init()
+        };
+        Ok(RenderModelPropertiesFB {
+            vendor_id: out.vendor_id,
+            model_name: unsafe { fixed_str(&out.model_name).into() },
+            model_key: out.model_key,
+            model_version: out.model_version,
+            flags: out.flags,
+        })
+    }
+
+    pub fn load_render_model_fb(&self, model_key: RenderModelKeyFB) -> Result<Vec<u8>> {
+        let mut buffer = sys::RenderModelBufferFB {
+            ty: sys::RenderModelBufferFB::TYPE,
+            next: ptr::null_mut(),
+            buffer_capacity_input: 0,
+            buffer_count_output: 0,
+            buffer: ptr::null_mut(),
+        };
+        unsafe {
+            cvt((self.instance().render_model_fb().load_render_model)(
+                self.as_raw(),
+                &RenderModelLoadInfoFB {
+                    ty: RenderModelLoadInfoFB::TYPE,
+                    next: ptr::null_mut(),
+                    model_key,
+                },
+                &mut buffer,
+            ))?;
+            let mut out = Vec::with_capacity(buffer.buffer_count_output as usize);
+            loop {
```

Comment:
Could this use `get_arr`?

Reply:
check it now, i switched to using get_arr